File: 01_Language/01_Functions/python/array_product.py
# coding: utf-8

import functools


def array_product(arr):
    if len(arr) == 0:
        return 1
    return functools.reduce(lambda x, y: x * y, arr)


if __name__ == '__main__':
    l = [2, 4, 6, 8]
    print(array_product(l))
    print(array_product([]))

File: 01_Language/01_Functions/js/array_product.js
function array_product(arr) {
    let res = 1;
    for (let i in arr) {
        res *= arr[i]
    }
    return res
}

let l = [2, 4, 6, 8]
console.log(array_product(l));
console.log(array_product([]));
